pkg/kubelet/kubelet.go

```diff
@@ -1041,6 +1057,64 @@ type Kubelet struct {
 	dockerLegacyService dockershim.DockerLegacyService
 }
 
+func initializeServerCertificateManager(kubeClient clientset.Interface, kubeCfg *componentconfig.KubeletConfiguration, nodeName types.NodeName, hostname string) (certificate.Manager, error) {
```

Comment:
see full (crazy) logic for determining the addresses to report in node status at https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet_node_status.go#L413

We should make sure that the reported addresses (IP and DNS names) are included as SANs in the cert